### ISSUE TITLE
Ensure extension works in Firefox when setting enabled to delete cookie and site data

### ISSUE THREAD

**Issue by john-light** [leather-io/extension#528]
The Blockstack Connect [announcement blog post](https://blog.blockstack.org/upgrade-your-app-onboarding-experience-with-blockstack-connect/) says:

> Of course, Blockstack Connect, the Blockstack App for authentication and its extension are all open-source. The extension is available for installation from source for Chrome or Firefox. We’ve also submitted it to the Chrome Web Store and Add-ons for Firefox for availability soon upon approval.
> 
> Blockstack Connect and the new authenticator have been active in beta testing since early this year, with today marking the official general release. You can try out BlockSurvey, Sigle, and Runkod for a taste of the new experience within the context of live apps built on the platform.

I installed the extension on Firefox and tried each of the apps mentioned (BlockSurvey, Sigle, and Runkod) and was not able to use any of these apps with the Firefox extension.

See video for a demo: https://www.youtube.com/watch?v=GzGjB7K5HGg

**Comment by markmhendrickson** [comment]
Thanks for this report, @john-light, and sorry you've run into this. 

Mind providing console logs for the originating app window in case there's something indicative in there? As well as information about your system and browser (such as add-ons that might be doing something here to modify inter-window behavior) that'd be helpful for recreation?

@timstackblock As you'll see in the screen grab, a new tab is getting opened with the app every time an account is selected for app authentication. At first glance, it appears something funky is going on with our callback to the original window. Let's see if we can recreate then diagnose.

cc @hstove

**Comment by hstove** [comment]
Hey @john-light , I see that the name you're using `johnlight2020.id` is not yet fully registered at the blockchain level. When did you register this name?

Also, thank you so much for opening this issue. You've been hugely helpful with the bugs and other issues you've brought up lately.

**Comment by john-light** [comment]
Hi folks, I went back to try again so I could see if the web console showed anything useful, and whatever was a blocker before no longer seems to be an issue, at least on Sigle and BlockSurvey. It's a bit strange to me since nothing has changed about my browsing setup but can't complain that things now work as expected!

However, I tried the extension in the same browser (Firefox 78.0.2) on another machine running the same OS (Linux Mint 18.3) and ran into the same problem. There's a setting in Firefox that I only had enabled on this second browser, which is `Delete cookies and site data when Firefox is closed`. After disabling this setting, the problem was solved and I could log in to Sigle. Is there a way to get the extension to work even with this setting enabled? It would be an annoyance to have to enable/disable it every time I want to use Blockstack, which is the only extension affected by this setting.

(Also I did find another bug but I'll create a separate issue for that.)

**Comment by markmhendrickson** [comment]
I'm glad to hear you got unblocked on that first machine, though that is a bit disconcerting that it went away without any intervention, suggesting a possible intermittent bug. We'll have to keep our eyes out.

As for the second machine, I'm re-titling this issue to "Ensure extension works in Firefox when setting enabled to delete cookie and site data" so we can look into that particular conflict. To confirm, you're unable to authenticate at all with the extension when this setting is enabled? It's not simply that your session is lost when restarting Firefox?

Please do file a separate issue for the other bug and we can follow up on that one, too. 🙏

**Comment by john-light** [comment]
> To confirm, you're unable to authenticate at all with the extension when this setting is enabled? It's not simply that your session is lost when restarting Firefox?

Correct. The Blockstack App displays similar behavior to that shown in the video in my OP on this issue when this setting is enabled. When I disable this setting, Blockstack App authentication works as expected.

**Comment by timstackblock** [comment]
I am able to reproduce this issue, thanks @john-light
